app/components/click_observer_component.rb

```diff
@@ -1,14 +1,22 @@
 # frozen_string_literal: true
 
 class ClickObserverComponent < BaseComponent
-  attr_reader :event_name, :tag_options
+  attr_reader :event_name, :tag_options, :payload
 
-  def initialize(event_name:, **tag_options)
+  def initialize(event_name:, payload: {}, **tag_options)
     @event_name = event_name
+    if payload.is_a?(Hash)
+      @payload = payload
+    else
+      @payload = {}
+    end
     @tag_options = tag_options
   end
 
   def call
-    content_tag(:'lg-click-observer', content, 'event-name': @event_name, **tag_options)
+    content_tag(
+      :'lg-click-observer', content, 'event-name': @event_name,
+                                     data: { payload: payload }, **tag_options
```

Comment:
This spacing and argument arrangement is a bit difficult to follow. I'd recommend one per line.

Also, the way this is implemented will prevent someone from being able to pass both `payload` as well as their own `data` attributes (`ClickObserverComponent.new(data: { foo: 'bar' }, payload: 'abc')`). We'll need `data` to come last and merge with `tag_options`. Can you add test coverage?

```suggestion
      :'lg-click-observer',
      content,
      'event-name': @event_name,
      **tag_options,
      data: { **tag_options[:data].to_h, payload: },
```